core_lib/src/interface/timelinecells.cpp

```diff
@@ -857,14 +903,29 @@ void TimeLineCells::mouseReleaseEvent(QMouseEvent* event)
     primaryButton = Qt::NoButton;
     mEndY = mStartY;
     mTimeLine->scrubbing = false;
+
     int frameNumber = getFrameNumber(event->pos().x());
-    if (frameNumber < 1) frameNumber = -1;
+    if (frameNumber < 1) frameNumber = 1;
     int layerNumber = getLayerNumber(event->pos().y());
-    if (mType == TIMELINE_CELL_TYPE::Tracks && primaryButton != Qt::MidButton && layerNumber != -1 && layerNumber < mEditor->object()->getLayerCount())
+    if (mType == TIMELINE_CELL_TYPE::Tracks && primaryButton != Qt::MidButton)
     {
-        Layer *currentLayer = mEditor->object()->getLayer(layerNumber);
+        // We should affect the current layer based on what's selected, not where the mouse currently is.
+        Layer *currentLayer = mEditor->layers()->getLayer(mCurrentLayerNumber);
+        if (currentLayer == nullptr) { return; }
```

Comment:
This return statement prevents later code in this method from running. Is that intentional?

Reply:
Hmm most likely not.. that seems to be an afterthought. Will change.